src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -121,46 +128,123 @@ public bool GetWordVector(ref DvText word, float[] wordVector)
             }
         }
 
+        /// <summary>
+        /// Information for each column pair.
+        /// </summary>
+        public sealed class ColumnInfo
+        {
+            public readonly string Input;
+            public readonly string Output;
+
+            public ColumnInfo(string input, string output)
+            {
+                Contracts.CheckNonEmpty(input, nameof(input));
+                Contracts.CheckNonEmpty(output, nameof(output));
+
+                Input = input;
+                Output = output;
+            }
+        }
+
         private const string RegistrationName = "WordEmbeddings";
 
         private const int Timeout = 10 * 60 * 1000;
 
         /// <summary>
-        /// Public constructor corresponding to <see cref="SignatureDataTransform"/>.
+        /// Instantiates <see cref="WordEmbeddingsTransform"/> using the pretrained word embedding model specified by <paramref name="modelKind"/>.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="inputColumn">Name of the input column.</param>
+        /// <param name="outputColumn">Name of the output column.</param>
+        /// <param name="modelKind">The pretrained word embedding model.</param>
+        public WordEmbeddingsTransform(IHostEnvironment env, string inputColumn, string outputColumn,
+           PretrainedModelKind modelKind = PretrainedModelKind.Sswe)
+           : this(env, modelKind, new ColumnInfo(inputColumn, outputColumn))
+        {
+        }
+
+        /// <summary>
+        /// Instantiates <see cref="WordEmbeddingsTransform"/> using the custom word embedding model by loading it from the file specified by the <paramref name="customModelFile"/>.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="inputColumn">Name of the input column.</param>
+        /// <param name="outputColumn">Name of the output column.</param>
+        /// <param name="customModelFile">Filename for custom word embedding model.</param>
+        public WordEmbeddingsTransform(IHostEnvironment env, string inputColumn, string outputColumn, string customModelFile)
```

Comment:
Is there a way to take in multiple columns (and produce multiple columns)?

Reply:
Please look at the constructors with `ColumnInfo` argument.

---
In reply to: [218261797](https://github.com/dotnet/machinelearning/pull/928#discussion_r218261797) [](ancestors = 218261797)

Reply:
Ahh great. Thanks.